### PR TITLE
fix(pkg/util/slices): constrain RemoveDuplicateElement to comparable types

### PR DESCRIPTION
**Pull request by Norway-02**
### What type of PR is this?
/kind bug
/kind cleanup

### What this PR does / why we need it:
`RemoveDuplicateElement[T any](slice []T) []T` in `pkg/util/slices/slices.go` currently uses an internal `map[any]struct{}` for tracking duplicate entries. If passed a slice containing non-comparable generic types (such as slices, maps, or structs containing slices), Go's runtime panics with `runtime error: hash of unhashable type`.

This PR refines the generic type parameter constraint from `[T any]` to `[T comparable]` and updates the tracking map to `map[T]struct{}`. This enforces type safety at compile time and prevents runtime panics.

Additionally, this PR expands unit tests in `slices_test.go` to cover integers, custom comparable structs, and edge cases, achieving 100% statement test coverage.

### Which issue(s) this PR fixes:
Fixes # NONE

### Special notes for your reviewer:
None. All package unit tests pass cleanly with 100% statement coverage.

### Does this PR introduce a user-facing change?:
```release-note
NONE
```

### CODE REVIEW

**Comment by Norway-02** [comment]
Hi @DoisLONG, could you please review this PR when you get a chance? It refines generic type constraints in `pkg/util/slices` for compile-time safety and expands unit test coverage to 100%. Thank you!

**Comment by Norway-02** [comment]
/assign @DoisLONG

Hi @DoisLONG @WillardHu, this PR updates generic type constraints in `pkg/util/slices` for compile-time safety and achieves 100% test coverage. Could you please review when you get a chance? Thank you!

File: pkg/metaserver/application.go [hunk 1]
```diff
@@ -158,8 +157,12 @@ func (a *Application) Namespace() string {
 }
 
 func (a *Application) Cancel() {
-	if a.cancel != nil {
-		a.cancel()
+	a.countLock.Lock()
```
Comment (WillardHu) [inline pkg/metaserver/application.go]: The Golang Context is designed to be thread-safe, so it does not require locks, and the `countLock` lock is designed to ensure the thread safety of the `count` field.

**Comment by Norway-02** [comment]
Hi @WillardHu, thank you for the review!

I have rebased the PR onto the latest master branch so that it contains only the single focused commit for `pkg/util/slices` type safety (`RemoveDuplicateElement[T comparable]`) and removed the unrelated commit.

All package unit tests pass cleanly with 100% statement coverage (`go test -v -race -cover ./pkg/util/slices/...`).

Could you please review again when you get a chance? Thank you!

**Comment by Norway-02** [comment]
Hi @WillardHu, all unit tests and core checks pass cleanly with 100% statement coverage for `pkg/util/slices`. The single CRI-O e2e job failed due to a transient network timeout during dependency fetch (`generate_mapper.sh`). Could you please review and `/approve` (or trigger `/retest`) when you get a chance so tide can merge it? Thank you!
